fofix/core/GameEngine.py

```diff
@@ -292,29 +286,96 @@ def __init__(self, config = None):
+
+        log.debug("Ready.")
+
+    # evilynux - This stops the crowd cheers if they're still playing (issue 317).
+    def quit(self):
+        # evilynux - self.audio.close() crashes when we attempt to restart
+        if not self.restartRequested:
+            self.audio.close()
+        Player.savePlayers()
+        for taskData in list(self.task.tasks):
+            self.task.removeTask(taskData['task'])
+        self.running = False
+
+    def _initTheme(self, themename, themepath):
+        """
+        Select the source of graphics for the game.
+        Note that currently this can only be called GameEngine on startup.
+
+        :param themename: what to call this theme
+        :type themename: str
+        :param themepath: absolute path to theme folder
+        :type themepath: str
+        """
+
+        log.notice( 'Setting theme %s from "%s"' % (themename,themepath) )
+
+        self.theme = None
+        try:
+            # Look for "CustomTheme.py" inside theme dir
+            fp, pathname, description = imp.find_module("CustomTheme",[themepath])
+            try:
+                # Found it! Load it.
+                theme = imp.load_module("CustomTheme", fp, pathname, description)
+                self.theme = theme.CustomTheme(themepath, themename)
+                log.notice('Theme activated using custom class "%s"' % pathname)
+            except ImportError as e:
+                # Unable to load module; log it, but continue with default Theme.
+                log.error('Failed to load CustomTheme.py from "%s"' % pathname)
+            finally:
+                fp.close()
+        except ImportError:
+            # CustomTheme is optional; no need to complain.
+            pass
+
+        if self.theme is None:
+            self.theme = Theme(themepath, themename)
+            log.notice("Theme activated using built-in Theme class")
+        self.task.addTask(self.theme)
+
+    def _initStages(self):
+        """
+        Setup animated stages from already-setup theme.
+        Only call this after _initTheme().
+
+        # <themename>\Stages still contains the backgrounds for when stage rotation is off, and practice.png
+        # subfolders under Stages\ will each be treated as a separate animated stage set
+        """
 
         self.stageFolders = []
-        currentTheme = themename
+        currentTheme = self.theme.name
 
-        stagespath = os.path.join(Version.dataPath(), "themes", currentTheme, "backgrounds")
-        themepath  = os.path.join(Version.dataPath(), "themes", currentTheme)
+        stagespath = os.path.join(themepath, "backgrounds")
         if os.path.exists(stagespath):
             self.stageFolders = []
-            allFolders = os.listdir(stagespath)   #this also includes all the stage files - so check to see if there is at least one .png file inside each folder to be sure it's an animated stage folder
+            allFolders = os.listdir(stagespath)
```

Comment:
`stagespath` is not defined anymore : you deleted it.

Reply:
Erm not sure what you mean here it is clearly defined above...

Reply:
One line above that if, in fact.